terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.ts

```diff
@@ -304,25 +309,97 @@ export async function getGHRunnerOrg(ec2runner: RunnerInfo, metrics: ScaleDownMe
     }
   }
 
-  if (ghRunner === undefined && ec2runner.ghRunnerId !== undefined) {
-    console.warn(
-      `Runner '${ec2runner.instanceId}' [${ec2runner.runnerType}](${org}) not found in ` +
-        `listGithubRunnersOrg call, attempting to grab directly`,
+  if (ghRunner === undefined) {
+    if (ec2runner.ghRunnerId === undefined) {
+      console.warn(
+        `Runner '${ec2runner.instanceId}' [${ec2runner.runnerType}](${org}) was neither found in ` +
+          `the list of runners returned by the listGithubRunnersOrg api call, nor did it have the ` +
+          `GithubRunnerId EC2 tag set.  This can happen if there's no runner running on the instance.`,
+      );
+    } else {
+      console.warn(
+        `Runner '${ec2runner.instanceId}' [${ec2runner.runnerType}](${org}) not found in ` +
+          `listGithubRunnersOrg call, attempting to grab directly`,
+      );
+      try {
+        ghRunner = await getRunnerOrg(ec2runner.org as string, ec2runner.ghRunnerId, metrics);
+      } catch (e) {
+        console.warn(
+          `Runner '${ec2runner.instanceId}' [${ec2runner.runnerType}](${org}) error when getRunnerOrg call: ${e}`,
+        );
+        /* istanbul ignore next */
+        if (isGHRateLimitError(e)) {
+          throw e;
+        }
+      }
+    }
+  } else if (dontTrustIdleFromList && !ghRunner.busy && ec2runner.ghRunnerId !== undefined) {
+    console.debug(
+      `Runner '${ec2runner.instanceId}' [${ec2runner.runnerType}](${org}) was found in the list of runners,` +
+        ` it should not be busy, but the flag dontTrustIdleFromList is set to true, so we will not trust it ` +
+        `and try to grab it directly if we have GHA quotas for it.`,
     );
+
+    let safeToCallGHApi = false;
     try {
-      ghRunner = await getRunnerOrg(ec2runner.org as string, ec2runner.ghRunnerId, metrics);
+      const ghLimitInfo = await getGitHubRateLimit({ owner: org, repo: '' }, metrics);
+      metrics.gitHubRateLimitStats(ghLimitInfo.limit, ghLimitInfo.remaining, ghLimitInfo.used);
+      if (ghLimitInfo.remaining > ghLimitInfo.limit * 0.4) {
```

Comment:
nit: can you please extract the 0.4 into a constant so that it's not a magic number in the code?